src/js/utils/dom.js

```diff
@@ -10,12 +10,29 @@ import document from 'global/document';
 import mejs from '../core/mejs';
 
 
-// Cuz ie sucks
-const TinyPromise = (h,f=[],b=-1,g,c,l) => (
-	l=d=>{for(g=d;c=f.shift();)c[b]&&c[b](d)},
-	h(d=>l(d,b=0),d=>l(d,b=1)),
-	{then(...d){~b?d[b]&&d[b](g):f.push(d)}}
-)
+function TinyPromise(handler) {
+	const thens = [];
+	let state = -1;
+	let result;
+	let then;
+
+	function done(value) {
+		for (result = value; then = thens.shift();) {
```

Comment:
When I run grunt it complain about this: `Expected a conditional expression and instead saw an assignment`, can you fix this?